clion/src/main/java/org/sonarlint/intellij/clion/AnalyzerConfiguration.java

```diff
@@ -175,19 +175,33 @@ static ForcedLanguage getSonarLanguage(OCLanguageKind languageKind) {
   }
 
   private boolean usingRemoteOrWslToolchain(OCResolveConfiguration configuration) {
-    final var initializedWorkspaces = CidrWorkspace.getInitializedWorkspaces(project);
-    CPPEnvironment cppEnvironment = null;
-    for (var initializedWorkspace : initializedWorkspaces) {
-      if (initializedWorkspace instanceof CMakeWorkspace cMakeWorkspace) {
-        cppEnvironment = getCMakeCppEnvironment(cMakeWorkspace, configuration);
-      } else {
-        cppEnvironment = tryReflection(initializedWorkspace);
-        if (cppEnvironment != null) {
-          break;
+    try {
+      tryIfClassAccessible();
+      final var initializedWorkspaces = CidrWorkspace.getInitializedWorkspaces(project);
+      for (var initializedWorkspace : initializedWorkspaces) {
+        if (Class.forName("com.jetbrains.cidr.cpp.cmake.workspace.CMakeWorkspace").isInstance(initializedWorkspace)) {
```

Comment:
Why do you need reflection here? Can't we assume this class is present if the 2 others are? It's also a change of behavior since it might throw, further `initializedWorkspaces` might not be checked

Reply:
Moved